src/hotspot/share/nmt/memReporter.cpp

```diff
@@ -432,34 +438,42 @@ void MemDetailReporter::report_virtual_memory_region(const ReservedMemoryRegion*
   }
 
   if (all_committed) {
-    CommittedRegionIterator itr = reserved_rgn->iterate_committed_regions();
-    const CommittedMemoryRegion* committed_rgn = itr.next();
-    if (committed_rgn->size() == reserved_rgn->size() && committed_rgn->call_stack()->equals(*stack)) {
-      // One region spanning the entire reserved region, with the same stack trace.
-      // Don't print this regions because the "reserved and committed" line above
-      // already indicates that the region is committed.
-      assert(itr.next() == nullptr, "Unexpectedly more than one regions");
+    bool reserved_and_committed = false;
+    VirtualMemoryTracker::Instance::tree()->visit_committed_regions(*reserved_rgn,
+                                                                  [&](CommittedMemoryRegion& committed_rgn) {
+      if (committed_rgn.size() == reserved_rgn->size() && committed_rgn.call_stack()->equals(*stack)) {
+        // One region spanning the entire reserved region, with the same stack trace.
+        // Don't print this regions because the "reserved and committed" line above
+        // already indicates that the region is committed.
+        reserved_and_committed = true;
+        return false;
+      }
+      return true;
+    });
+
+    if (reserved_and_committed)
       return;
-    }
   }
 
-  CommittedRegionIterator itr = reserved_rgn->iterate_committed_regions();
-  const CommittedMemoryRegion* committed_rgn;
-  while ((committed_rgn = itr.next()) != nullptr) {
+  auto print_committed_rgn = [&](CommittedMemoryRegion* crgn) {
```

Comment:
`crgn` can be const reference instead of pointer.

Reply:
Done.